platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -801,7 +810,8 @@ int _odp_ipsec_sa_stats_update(ipsec_sa_t *ipsec_sa, uint32_t len,
 		    packets >= ipsec_sa->hard_limit_packets)
 			sa_tl->lifetime_status.error.hard_exp_packets = 1;
 	}
-	sa_tl->packet_quota--;
+
+	odp_atomic_dec_u32(&sa_tl->packet_quota);
```

Comment:
Atomic incs and decs are actually unnecessarily strong here as this thread is the only writer. IOW, we do not need atomicity of the whole read-modify-write sequence but only atomicity of the store so that the stats query thread can load the value without a data race. Atomic load, followed by increment/decrement, followed by atomic store should be faster than an atomic inc/dec in many systems.

Reply:
Agreed. Will have this fixed in next version.